src/robusta/utils/silence_utils.py

```diff
@@ -0,0 +1,146 @@
+import json
+from datetime import datetime
+from enum import Enum
+from typing import Dict, List, Optional, Union
+from uuid import UUID
+
+import requests
+from pydantic import BaseModel, SecretStr, validator
+
+from robusta.core.model.base_params import ActionParams
+from robusta.integrations.prometheus.utils import ServiceDiscovery, AlertManagerDiscovery
+
+
+# ref to api https://github.com/prometheus/alertmanager/blob/main/api/v2/openapi.yaml
+
+
+class Matcher(BaseModel):
+    # https://github.com/prometheus/alertmanager/blob/main/api/v2/models/matcher.go
+    isEqual: bool = (
+        True  # support old version matchers with omitted isEqual https://github.com/prometheus/alertmanager/pull/2603
+    )
+    isRegex: bool
+    name: str
+    value: str
+
+
+class SilenceStatus(BaseModel):
+    # https://github.com/prometheus/alertmanager/blob/main/api/v2/models/silence_status.go
+    state: str
+
+
+class Silence(BaseModel):
+    # https://github.com/prometheus/alertmanager/blob/main/api/v2/models/silence.go
+    id: UUID
+    status: SilenceStatus
+    comment: str
+    createdBy: str
+    startsAt: datetime
+    endsAt: datetime
+    matchers: List[Matcher]
+
+    def to_list(self) -> List[str]:
+        return [
+            str(self.id),
+            self.status.json(),
+            self.comment,
+            self.createdBy,
+            self.startsAt.isoformat(timespec="seconds"),
+            self.endsAt.isoformat(timespec="seconds"),
+            json.dumps([matcher.dict() for matcher in self.matchers]),
+        ]
+
+
+class BaseSilenceParams(ActionParams):
+    """
+    :var alertmanager_url: Alternative Alert Manager url to send requests.
+    """
+
+    alertmanager_flavor: str = None  # type: ignore
+    alertmanager_url: Optional[str]
+    alertmanager_auth: Optional[SecretStr] = None
+    grafana_api_key: str = None  # type: ignore
+
+    @validator("alertmanager_url", allow_reuse=True)
+    def remove_ending_slash(cls, v):
+        if v.endswith("/"):
+            return v[:-1]
+        return v
+
+
+class DeleteSilenceParams(BaseSilenceParams):
+    """
+    :var id: uuid of the silence.
+    """
+
+    id: str
+
+
+class AddSilenceParams(BaseSilenceParams):
+    """
+    :var id: uuid of the silence. use for update, empty on create.
+    :var comment: text comment of the silence.
+    :var createdBy: author of the silence.
+    :var startsAt: date.
+    :var endsAt: date.
+    :var matchers: List of matchers to filter the silence effect.
+    """
+
+    id: Optional[str]
+    comment: str
+    createdBy: str
+    startsAt: datetime
+    endsAt: datetime
+    matchers: List[Matcher]
+
+
+def get_alertmanager_silences_connection(params: BaseSilenceParams) -> Optional[Union[requests.Response, requests.Response]]:
```

Comment:
The typing were for the connection `Response`.

Reply:
My question here was, because it was a Union of 2 types, which are the same `requests.Response`:
`Optional[Union[requests.Response, requests.Response]]`